### PR TITLE
Jenish/feature/nand to obc transfer

### PR DESCRIPTION
**Pull request by JenishPatel99**

### CODE REVIEW

File: Core/Src/command_handler.c [hunk 1]
```diff
@@ -73,21 +79,34 @@ void take_image() {
...
-    // TODO: Read number of images stored in NAND fs
-    *(image_count) = 0;
-}
+void get_image_count(uint8_t *cnt) { *(cnt) = image_count; }
 
 /**
- * @brief Get the image length
+ * @brief Get the image length fr
  *
  * @param image_length: Pointer to variable containing length of image
```
Comment (Taylorr82) [inline Core/Src/command_handler.c:87]: It's not really a pointer to the image length, it's a pointer to a return value. Though I think this function should instead return the length of the image or -1. -1 is not a valid length, so it's clearly an error

File: Core/Src/command_handler.c [hunk 2]
```diff
@@ -266,6 +285,92 @@ void get_rtc_time(Iris_Timestamp *timestamp) {
...
+
+    DBG_PUT("%d|%s|%d", image_file_infos_queue[image_count].file_id, image_file_infos_queue[image_count].file_name,
+            image_file_infos_queue[image_count].file_size);
+
+    image_count += 1;
+    return 0;
+}
+
+int delete_image_file_from_queue(uint16_t index) {
+    image_file_infos_queue[index].file_id = 0xFF;
```
Comment (Taylorr82) [inline Core/Src/command_handler.c:346]: you set file ID to 0xFF and then delete ID 0xFF?

File: Core/Src/main.c [hunk 3]
```diff
@@ -767,6 +769,53 @@ static void onboot_commands(void) {
...
+ * @brief Store information from existing files from NAND to
+ * 		  RAM (buffer)
+ *
+ * 		  The purpose of transferring file info (file_id, file_name,
+ * 		  file_size) from NAND to RAM during on-boot is to set up a
+ * 		  quick lookup table for image transfer APIs to quickly
+ * 		  extract file information, instead of accessing NAND structures
+ * 		  directly.
+ */
+static int store_file_infos_in_buffer() {
```
Comment (Taylorr82) [inline Core/Src/main.c:783]: Shouldn't this be in nandfs? I don't like how it uses an extern

File: Core/Src/command_handler.c (hunk 2 above)
Comment (JenishPatel99) [reply Core/Src/command_handler.c]: Yeah, that was a mistake. Should be fixed now

File: Core/Src/main.c (hunk 3 above)
Comment (JenishPatel99) [reply Core/Src/main.c]: Yeah, you correct. I have moved this the command_handler where the buffer is created

**Review by Taylorr82 — APPROVED** [review]

**Pull request merged**